### ISSUE TITLE
RangeDateFilter does not send ISO date format in query

### ISSUE THREAD

**Issue by VaZark** [unfoldadmin/django-unfold#491]
I noticed that the RangeDate filters do not work and the results fetched apply only the other filters.

After some digging, it looks like the widget and the query send `DD/MM/YYYY` instead of the `YYYY-MM-DD`. This in turn throws an exception `ValidationError(['Le format de la valeur «\xa004/06/2024\xa0» n’est pas valide. Le format correct est AAAA-MM-JJ HH:MM[:ss[.uuuuuu]][FH].'])`

Manually entering the date in the ISO Format in the input seems to work fine

Fix: Update the query string set to the input to respect the ISO Format (maybe use `type="date"`?)

**Comment by VaZark** [comment]
Adding `"type"="date"` makes the widget functional again but we lose the pretty calendar dialog

https://github.com/unfoldadmin/django-unfold/blob/b771b842e571679c90ee00ae6b25667f838c78ae/src/unfold/contrib/filters/forms.py#L98-L104

**Comment by PierreMory** [comment]
I confirm the original issue.

To reproduce this, just change the LANGUAGE_CODE setting of the Django project to "fr-fr".
`LANGUAGE_CODE = 'fr-fr'`

Then, if you apply a RangeDateFilter, it would generate the following URL (for example), which does not return any error but the filter doesn't work as the date is not formatted in ISO:
`https://localhost:8000/admin/auth/user/?last_login_from=23%2F09%2F2024`

**Comment by wilfredinni** [comment]
Same problem here with `es-cl`

**Comment by wllbll** [comment]
Same problem here with zh-Hans

**Comment by wllbll** [comment]
my dirty but works solutions, can preserves the pretty calendar dialog, just force to replace / with -

```python
class MyDirtyRangeDateFilter(RangeDateFilter):
    def queryset(self, request: HttpRequest, queryset: QuerySet) -> QuerySet:
        filters = {}

        value_from = self.used_parameters.get(self.parameter_name + "_from", None)
        if value_from not in EMPTY_VALUES:
            value_from = value_from.replace('/', '-')
            filters.update(
                {
                    # self.parameter_name + "__gte": self.used_parameters.get(
                    #     self.parameter_name + "_from", None
                    # ),
                    self.parameter_name + "__gte": value_from
                }
            )

        value_to = self.used_parameters.get(self.parameter_name + "_to", None)
        if value_to not in EMPTY_VALUES:
            value_to = value_to.replace('/', '-')
            filters.update(
                {
                    # self.parameter_name + "__lte": self.used_parameters.get(
                    #     self.parameter_name + "_to", None
                    # ),
                    self.parameter_name + "__lte": value_to
                }
            )

        try:
            return queryset.filter(**filters)
        except (ValueError, ValidationError):
            return None
```

**Comment by lukasvinclav** [comment]
Fixed in #857

It is important to properly configure settings below if you are using some special formats which are not listed there:

https://docs.djangoproject.com/en/5.1/ref/settings/#date-input-formats

https://docs.djangoproject.com/en/5.1/ref/settings/#datetime-input-formats
